src/angular/codemods/app-config.ts

```diff
@@ -0,0 +1,232 @@
+
+  if (!hasRouter) {
+    imports.$add({
+      from: '@angular/router',
+      imported: 'Router',
+      local: 'Router',
+    });
+  }
+}
+
+function addMissingImportsV19(
+  originalAppConfigMod: ProxifiedModule<any>,
+): void {
+  const imports = originalAppConfigMod.imports;
+
+  const hasProvideAppInitializer = imports.$items.some(
+    (item) => item.local === 'provideAppInitializer',
+  );
+
+  if (!hasProvideAppInitializer) {
+    imports.$add({
+      from: '@angular/core',
+      imported: 'provideAppInitializer',
+      local: 'provideAppInitializer',
+    });
+  }
+
+  const hasInject = imports.$items.some((item) => item.local === 'inject');
+
+  if (!hasInject) {
+    imports.$add({
+      from: '@angular/core',
+      imported: 'inject',
+      local: 'inject',
+    });
+  }
+}
+
+function addAppInitializer(originalAppConfigMod: ProxifiedModule<any>): void {
+  const imports = originalAppConfigMod.imports;
+
+  const hasAppInitializer = imports.$items.some(
+    (item) => item.local === 'APP_INITIALIZER',
+  );
+
+  if (!hasAppInitializer) {
+    imports.$add({
+      from: '@angular/core',
+      imported: 'APP_INITIALIZER',
+      local: 'APP_INITIALIZER',
+    });
+  }
+}
+
+function addImports(
+  originalAppConfigMod: ProxifiedModule<any>,
+  isAboveAngularV19: boolean,
+  isTracingEnabled: boolean,
+): void {
+  addSentryImport(originalAppConfigMod);
+  addErrorHandlerImport(originalAppConfigMod);
+  addRouterImport(originalAppConfigMod);
+
+  if (isAboveAngularV19) {
+    addMissingImportsV19(originalAppConfigMod);
+  } else if (isTracingEnabled) {
+    addAppInitializer(originalAppConfigMod);
+  }
+}
+
+function addProviders(
+  originalAppConfigMod: ProxifiedModule<any>,
+  isAboveAngularV19: boolean,
+  isTracingEnabled: boolean,
+): void {
+  const b = recast.types.builders;
+
+  recast.visit(originalAppConfigMod.exports.$ast, {
+    visitExportNamedDeclaration(path) {
+      // @ts-expect-error - declaration should always be present in this case
+      if (path.node.declaration.declarations[0].id.name === 'appConfig') {
+        const appConfigProps =
+          // @ts-expect-error - declaration should always be present in this case
+          path.node.declaration.declarations[0].init.properties;
+
+        const providers = appConfigProps.find(
+          (prop: ObjectProperty) =>
+            (prop.key as Identifier).name === 'providers',
+        ).value as ArrayExpression;
+
+        const errorHandlerObject = b.objectExpression([
+          b.objectProperty(
+            b.identifier('provide'),
+            b.identifier('ErrorHandler'),
+          ),
+          b.objectProperty(
+            b.identifier('useValue'),
+            b.identifier('Sentry.createErrorHandler()'),
+          ),
+        ]);
+
+        providers.elements.push(
+          // @ts-expect-error - errorHandlerObject is an objectExpression
+          errorHandlerObject,
+        );
```

Comment:
m: Angular only permits one ErrorHandler per application. We need to check if users already declared one and if so, let's not add an additional one but link to our docs how to combine their handler with ours: https://docs.sentry.io/platforms/javascript/guides/angular/features/error-handler/